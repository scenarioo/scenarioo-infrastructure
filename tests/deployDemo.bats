#!/usr/bin/env bats

load environment

setup() {
    setup_environment
}

@test "deployDemo - it should generate a correct <demo>.config" {
    result="$(./infra.sh deployDemo testBranch 227 testUser http://foobar/PR 700 | grep -v timestamp)"
    expected="$(cat tests/expected/deployDemo.out | grep -v timestamp)"
    # Check output
    diff <(echo "$result") <(echo "$expected")

    # Check JSON
    result="$(cat $DEMO_DIR/testBranch.json | grep -v timestamp)"
    expected="$(cat tests/expected/deployDemo_testBranch.json | grep -v timestamp)"
    diff <(echo "$result") <(echo "$expected")
}

@test "deployDemo - it should rotate builds by build number" {
    ./infra.sh deployDemo testBranch 227 testUser http://foobar/PR 700
    ./infra.sh deployDemo testBranch 230 testUser http://foobar/PR 700
    ./infra.sh deployDemo testBranch 232 testUser http://foobar/PR 700
    result="$(cat $OVERVIEW_TARGET_DIR/demos.json | grep -v timestamp)"
    expected="$(cat tests/expected/deployDemo_rotateBuild_demos.json | grep -v timestamp)"

    # Check aggregated JSON
    diff <(echo "$result") <(echo "$expected")
}


@test "deployDemo - it should rotate demos by timestamp" {
    ./infra.sh deployDemo demo1 227 testUser http://foobar/PR 700
    sleep 1 # With cached curl calls we might return here in under a second and end up with the same timestamp => sleep
    ./infra.sh deployDemo demo2 230 testUser http://foobar/PR 700
    sleep 1
    ./infra.sh deployDemo demo3 232 testUser http://foobar/PR 700
    result="$(cat $OVERVIEW_TARGET_DIR/demos.json | grep -v timestamp)"
    expected="$(cat tests/expected/deployDemo_rotateDemos_demos.json | grep -v timestamp)"
    cp $OVERVIEW_TARGET_DIR/demos.json tests
    # Check aggregated JSON
    diff <(echo "$result") <(echo "$expected")
}


teardown() {
    restore_environment
}