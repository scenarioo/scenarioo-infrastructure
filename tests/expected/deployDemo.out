WARNING: CIRLCE_TOKEN is not set. Without it we can't fetch artifacts from builds!
Getting list of artifacts ...
Getting WAR
Getting E2E docu
Getting example docu
New branch config:
{
  "branchName": "testBranch",
  "encodedBranchName": "testBranch",
  "timestamp": "1541188077",
  "buildNumber": "227",
  "triggeredBy": "testUser",
  "pullRequestUrl": "http://foobar/PR",
  "buildUrl": "https://circleci.com/gh/scenarioo/scenarioo/227",
  "warArtifact": "https://227-10269178-gh.circle-artifacts.com/0/scenarioo.war",
  "warArtifactSha256": "5cbda4b2145d5a78d695a9a01f37717f2fdb0f69436ad55a28c4a0a00a1664cc",
  "exampleDocuArtifact": "https://227-10269178-gh.circle-artifacts.com/0/exampleScenariooDocu.zip",
  "exampleDocuArtifactSha256": "420d68a8e37e62c19d0b6af47bbb658c673c31746b651d6f90c23b2e5230e416",
  "docuArtifacts": [
    {
      "url": "https://227-10269178-gh.circle-artifacts.com/0/e2eScenariooDocu.zip",
      "sha256": "321cf783e6b740b33e69735b513d8aa337bfc45482804162f3b3f39a84b6490e",
      "build": "227"
    }
  ]
}
Writing config file tests/tmp/demos/testBranch.json...
Demo testBranch added. Starting cleanup ...

Updating list of demos: overviewpage/demos.json
Persistent branches: "master","develop"
Max demos:       2
Total demos:     1
Demos to remove: 0
==================================
==================================
Updating list of demos: overviewpage/demos.json
